nautilus_trader/core/datetime.pyx

```diff
@@ -301,3 +301,29 @@ cpdef str format_iso8601(datetime dt):
 
     cdef tuple dt_partitioned = tz_stripped.rpartition('.')
     return f"{dt_partitioned[0]}.{dt_partitioned[2][:3]}Z"
+
+
+def max_date(date1: str | int | None = None, date2: str | int | None = None) -> str | None:
```

Comment:
Should probably either return a `date` object or append the name with `_str` or similar to make this clearer.
Also, this function is actually handling timestamps?

Reply:
In the code base timestamp is what is used to handle strings and ints indifferently as dates at config level. I decided to return a string so it's easier to debug at config/node level as dates are still strings there that will be converted again to timestamps.